pkg/datastores/container/store.go

```diff
@@ -0,0 +1,123 @@
+package container
+
+import (
+	"time"
+
+	"github.com/aquasecurity/tracee/api/v1beta1/datastores"
+)
+
+// DataStore interface implementation
+
+// Name returns the name of this datastore
+func (m *Manager) Name() string {
+	return "container"
+}
+
+// GetHealth returns the current health status of the datastore
+func (m *Manager) GetHealth() *datastores.HealthInfo {
+	// Try to acquire read lock with timeout to detect deadlocks
+	lockAcquired := make(chan struct{})
+	go func() {
+		m.lock.RLock()
+		_ = len(m.containerMap) // Read data to avoid empty critical section
+		m.lock.RUnlock()
+		close(lockAcquired)
+	}()
+
+	select {
+	case <-lockAcquired:
+		return &datastores.HealthInfo{
+			Status:    datastores.HealthHealthy,
+			Message:   "",
+			LastCheck: time.Now(),
+		}
+	case <-time.After(100 * time.Millisecond):
+		return &datastores.HealthInfo{
+			Status:    datastores.HealthUnhealthy,
+			Message:   "lock acquisition timeout - possible deadlock",
+			LastCheck: time.Now(),
+		}
+	}
```

Comment:
besides not linking this method, if we decide to add it, we have a possible go routine leak here in case the lock can't get acquired, every time the method is called, we create a go routine, if we can't acquire the lock the timeout fires and returns unhealthy, and the go routine trying to acquire the lock never ends, keeps running and trying to acquire it